File: api-course/src/main/java/fr/couture/course/entity/Categorie.java
package fr.couture.course.entity;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode
public class Categorie implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long ID;

    @Column(unique = true, nullable = false)
    private String nom;

    @OneToMany(orphanRemoval = true, mappedBy = "categorie")
    private List<Produit> produits = List.of();
}

File: front-course/src/app/core/models/categorie.ts
import {Produit} from "./produit";

export class Categorie {
  public id: number;
  public nom: string;
  public produits: Produit[];

  constructor(id: number, nom: string, produits: Produit[]) {
    this.id = id;
    this.nom = nom;
    this.produits = produits;
  }
}
